backend.native/tests/runtime/workers/worker4.kt

```diff
@@ -9,14 +9,65 @@ import kotlin.test.*
 
 import kotlin.native.concurrent.*
 
-@Test fun runTest() {
-    val worker = Worker.start()
-    val future = worker.execute(TransferMode.SAFE, { 41 }) {
-        input -> input + 1
+@Test fun runTest1() {
+    withWorker {
+        val future = execute(TransferMode.SAFE, { 41 }) { input ->
+            input + 1
+        }
+        future.consume { result ->
+            println("Got $result")
+        }
     }
-    future.consume {
-        result -> println("Got $result")
+    println("OK")
+}
+
+@Test fun runTest2() {
+    withWorker {
+        val counter = AtomicInt(0)
+
+        executeAfter(0, {
+            assertTrue(Worker.current.park(10_000_000_000, false))
+            assertEquals(counter.value, 0)
+            assertTrue(Worker.current.processQueue())
+            assertEquals(1, counter.value)
+            // Let main proceed.
+            counter.increment()  // counter becomes 2 here.
+            assertTrue(Worker.current.park(10_000_000_000, true))
+            assertEquals(3, counter.value)
+        }.freeze())
+
+        executeAfter(0, {
+            counter.increment()
+        }.freeze())
+
+        while (counter.value == 1) {
+            Worker.current.park(1_000_000)
```

Comment:
This never happens.

Reply:
No, it could be either 0 or 1, depending on if `assertTrue(Worker.current.processQueue())` executed job with increment scheduled line above or not.